docker.go

```diff
@@ -514,19 +516,76 @@ type DockerProvider struct {
 
 var _ ContainerProvider = (*DockerProvider)(nil)
 
+// or through Decode
+type TestContainersConfig struct {
+	Host      string `properties:"docker.host,default="`
+	TLSVerify int    `properties:"docker.tls.verify,default=0"`
+	CertPath  string `properties:"docker.cert.path,default="`
+}
+
 // NewDockerProvider creates a Docker provider with the EnvClient
 func NewDockerProvider() (*DockerProvider, error) {
-	client, err := client.NewEnvClient()
+	tcConfig := readTCPropsFile()
+	host := tcConfig.Host
+
+	opts := []client.Opt{client.FromEnv}
+	if host != "" {
```

Comment:
if dockerhost is not set, we should ignore other settings (e.g. TLS), as they come together with dockerhost and make no sense without it, so it can be one big `if` :)

Reply:
Addressed in https://github.com/testcontainers/testcontainers-go/pull/373/commits/59a295c5c337a81a41d8b4da211eb22974555f24